source/features/deep-blame.tsx

```diff
@@ -0,0 +1,111 @@
+import './deep-blame.css';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import React from 'dom-chef';
+import features from '../libs/features';
+import {getRepoGQL, getReference} from '../libs/utils';
+import delegate, {DelegateEvent} from 'delegate-it';
+import versionIcon from 'octicon/versions.svg';
+import octofaceIcon from 'octicon/octoface.svg';
+
+const getPullRequestBlameCommit = async (commit: string): Promise<string | undefined> => {
+	const {repository} = await api.v4(`
+		repository(${getRepoGQL()}) {
+			object(expression: "${commit}") {
+				... on Commit {
+					associatedPullRequests(last: 1) {
+						nodes {
+							mergeCommit {
+								oid
+							}
+							commits(last: 1) {
+								nodes {
+									commit {
+										oid
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	`);
+
+	const {nodes} = repository.object.associatedPullRequests;
+	const mergeCommit = nodes[0].mergeCommit.oid;
+
+	if (mergeCommit === commit) {
+		return nodes[0].commits.nodes[0].commit.oid;
+	}
+
+	return undefined;
+};
+
+function getCommitHash(commit: HTMLElement): string {
+	return (commit.nextElementSibling! as HTMLAnchorElement).href.split('/').slice(-1)[0];
+}
+
+async function redirectToBlameCommit(event: DelegateEvent<MouseEvent, HTMLLinkElement>): Promise<void | false> {
+	const {lineNumber, commit: pullRequestCommit} = event.delegateTarget.dataset;
+	if (event.delegateTarget.href && event.altKey) {
+		event.preventDefault();
+	}
+
+	event.delegateTarget.classList.add('anim-pulse');
+	event.delegateTarget.replaceChild(octofaceIcon(), event.delegateTarget.firstElementChild!);
+
+	const prBlameCommit = await getPullRequestBlameCommit(String(pullRequestCommit));
+	if (!prBlameCommit) {
+		event.delegateTarget.removeChild(event.delegateTarget.firstElementChild!);
```

Comment:
Use .remove and .replaceWith, never “*Child” APIs. 

Also there should be an `alert` in here

Reply:
I am working on this now. It should be a window alert?

Reply:
@fregante I just realized how is this case even possible.

Are we supposed to check the linked PR number? Since The merge commit will always = a merge commit but it may not be the correct PR.

Reply:
Ohhh Now i remember.... Ouch when I rewrote the code I deleted the commit to check against LOL

Reply:
Commit titles are custom:

```
Fix dropdown (replaces PR #1234)
```

Definitely an edge case. An alternative would be to just skip the check and just link to the PR; this would also simplify the code and wouldn't leave the user hanging... even if technically it's wrong... but that's what users get for mentioning unrelated PRs.

Reply:
I am not sure. I am so lost. LOL
I think that leading the user to the wrong place is worse. 
There is another case, where there was force push and the Pr Ccommit was replaced like [this](https://github.community/t5/GitHub-API-Development-and/assiociatedPullRequests-are-lost-after-history-rewrite/m-p/26950#M2187) idiot

Reply:
> like [this](https://github.community/t5/GitHub-API-Development-and/assiociatedPullRequests-are-lost-after-history-rewrite/m-p/26950#M2187)

In that case, a more lax version would actually work for them, especially if we just query the PR's last commit instead of using `associatedPullRequests`

If we use `associatedPullRequests`, that won't work regardless of this suggested change.

Reply:
We can leave it as is, with the `alert`

Reply:
> I am not sure. I am so lost. LOL

You can slow down